pkg/codegen/schema/schema.go

```diff
@@ -911,14 +931,30 @@ func (t *types) bindType(spec TypeSpec) (Type, error) {
 		}
 
 		// Parse the ref and look up the type in the type map.
-		if !strings.HasPrefix(spec.Ref, "#/types/") {
-			return nil, errors.Errorf("failed to parse ref %s", spec.Ref)
+		var token string
+		isResource := false
+		var err error
+		switch {
+		case strings.HasPrefix(spec.Ref, "#/types/"):
+			token, err = url.PathUnescape(spec.Ref[len("#/types/"):])
+		case strings.HasPrefix(spec.Ref, "#/resources/"):
+			token, err = url.PathUnescape(spec.Ref[len("#/resources/"):])
+			isResource = true
+		default:
+			err = fmt.Errorf("unknown ref")
 		}
-
-		token, err := url.PathUnescape(spec.Ref[len("#/types/"):])
 		if err != nil {
 			return nil, errors.Errorf("failed to parse ref %s", spec.Ref)
 		}
+
+		if isResource {
+			typ, ok := t.resources[token]
+			if !ok {
+				typ = &ResourceType{Token: token}
+				t.resources[token] = typ
+			}
+			return typ, nil
+		}
```

Comment:
Might as well move this up into the `case` and eliminate the extra variable + branch. You could move the token-type-related code into its branch, too, for consistency.